src/main/java/slimeknights/tconstruct/library/recipe/casting/ContainerFillingRecipeBuilder.java

```diff
@@ -0,0 +1,102 @@
+package slimeknights.tconstruct.library.recipe.casting;
+
+import com.google.gson.JsonObject;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import net.minecraft.advancements.Advancement;
+import net.minecraft.data.IFinishedRecipe;
+import net.minecraft.item.Item;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.util.IItemProvider;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.registries.ForgeRegistries;
+import slimeknights.tconstruct.library.recipe.AbstractRecipeBuilder;
+import slimeknights.tconstruct.smeltery.TinkerSmeltery;
+
+import javax.annotation.Nullable;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+public class ContainerFillingRecipeBuilder extends AbstractRecipeBuilder<ContainerFillingRecipeBuilder> {
+
+  private final ContainerFillingRecipeSerializer<?> recipeSerializer;
+  private String group;
+  private int fluidAmount = 0;
+  private final Item result;
+
+  private ContainerFillingRecipeBuilder(IItemProvider result, ContainerFillingRecipeSerializer<?> recipeSerializer) {
+    this.result = result.asItem();
+    this.recipeSerializer = recipeSerializer;
+  }
+
+  public static ContainerFillingRecipeBuilder castingRecipe(IItemProvider result, ContainerFillingRecipeSerializer<?> recipeSerializer) {
+    return new ContainerFillingRecipeBuilder(result, recipeSerializer);
+  }
+
+  public static ContainerFillingRecipeBuilder basinRecipe(IItemProvider result) {
+    return castingRecipe(result, TinkerSmeltery.basinFillingRecipeSerializer.get());
+  }
+
+  public static ContainerFillingRecipeBuilder tableRecipe(IItemProvider result) {
+    return castingRecipe(result, TinkerSmeltery.tableFillingRecipeSerializer.get());
+  }
+
+  public ContainerFillingRecipeBuilder setGroup(String group) {
+    this.group = group;
+    return this;
+  }
+
+  public ContainerFillingRecipeBuilder setFluidAmount(int fluidAmount) {
```

Comment:
Could probably put the fluid amount in the main constructor. Furnace recipes have four arguments in the main constructor, excluding the serializer.

Reply:
True enough. So I could prolly nix this method entirely?

Reply:
I think it would be cleaner to do so. Means no chance the user forgets a fluid amount.